sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/kafka/BindingServicePropertiesBeanPostProcessor.java

```diff
@@ -39,6 +39,7 @@ public Object postProcessBeforeInitialization(Object bean, String beanName) thro
 
                 BinderProperties defaultKafkaBinder = new BinderProperties();
                 defaultKafkaBinder.setEnvironment(environment);
+                defaultKafkaBinder.setDefaultCandidate(!StringUtils.hasText(bindingServiceProperties.getDefaultBinder()));
```

Comment:
Can you explain why this would resolve the issue?

Reply:
The `BindingServiceConfiguration` will depend on the properties `BindingServiceProperties`, and our processor [BindingServicePropertiesBeanPostProcessor#postProcessBeforeInitialization](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/kafka/BindingServicePropertiesBeanPostProcessor.java#L41) will always set a default kafka binder and the property `defaultCandidate` is true, it results the [Map<String, BinderConfiguration> getBinderConfigurations()](https://github.com/spring-cloud/spring-cloud-stream/blob/fe1392efb0e9f725b2e1ff360d3e3d4ceaf1a4db/core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceConfiguration.java#L93) method returns only one binder processor added, even through user has configured other binders. Finaly the initiated `DefaultBinderFactory` has custom default binder `servicebus` and its binder configurations do not contain the service bus binder.

![image](https://github.com/user-attachments/assets/db944928-fb08-4475-8d46-0f5840385e92)

Reply:
Per offline sync, we'd better modify the logic in the env post processor, to not register a default binder when there's no binders defined in application.yaml files and the specified default binder is set to another value.